specs/Standards/Tokens/NonFungibleToken/Event.md

```diff
@@ -66,6 +66,13 @@ interface NftTransferLog {
     token_ids: string[],
     memo?: string
 }
+
+// An event log to capture contract metadata updates. Note that the updated contract metadata is not included in the log, as it could easily exceed the 16KB log size limit. Listeners can query `nft_metadata` to get the updated contract metadata.
+// Arguments
+// * `memo`: optional message
+interface NftContractMetadataUpdateLog {
```

Comment:
We need to track NFT metadata (for one token) updates as well, I think it's better to support it in the same extension.
We may add optional `token_id` field here for that (and rename the structure), or add separate `NftMetadataUpdateLog` structure

Reply:
BTW, it should be possible to batch this. So it should probably be `token_ids`
In this case, I vote for the separate structure, it would be cleaner

Reply:
hey @telezhnaya, thanks for the comments! I totally agree that we need to track individual NFT metadata updates as well, and I originally proposed these both as part of the same extension, but the proposal of individual NFT metadata updates opened a whole can of worms ("dynamic" NFTs etc) so it was decided to separate these into their own PRs so that we can push this one through faster, as it is urgently needed.

See original discussion here: https://gov.near.org/t/extending-nep-171-events-standard-to-include-nft-update-events/30285/6

Reply:
Thanks for the link @lachlanglen !
We need to think how it's better to support such cases. 
We still may want to track such updates, that will give us the ability to show the evolution of the NFT, so we anyway may need the logs for that. 
Or at least we may want to know if the NFT is dynamic or not, it requires one more extension, adding the flag to the metadata. And maybe some other information such as the frequency of the updates, etc. We definitely need to research it deeper.
FYI @pkudinov 

Anyway, I fully agree it's the issue for the separate PR.